### PR TITLE
Async setup for Event Schemas.

### PR DESCRIPTION
**Pull request by achettyiitr**
**Fixes** # (*issue*)

## Type of change
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update
## Checklist:
- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have added unit tests for the code
- [ ] I have made corresponding changes to the documentation

### CODE REVIEW

File: event-schema/event_schema.go [hunk 1]
```diff
@@ -939,11 +950,17 @@ func (manager *EventSchemaManagerT) populateSchemaVersion(o *OffloadedSchemaVers
 
 // This should be called during the Initialize() to populate existing event Schemas
 func (manager *EventSchemaManagerT) populateEventSchemas() {
+	defer setEventSchemasPopulated(true)
```
Comment (psrikanth88) [inline event-schema/event_schema.go:953]: can we add a metric around how much time this step is taking

File: event-schema/event_schema.go [hunk 2]
```diff
@@ -710,6 +716,11 @@ func eventTypeIdentifier(eventType, eventIdentifier string) string {
 
 func (manager *EventSchemaManagerT) offloadEventSchemas() {
 	for {
+		if !areEventSchemasPopulated {
+			time.Sleep(time.Duration(10))
```
Comment (psrikanth88) [inline event-schema/event_schema.go:720]: time.Duration(10) would be 10 nano seconds.  Make it something like time.Sleep(1*time.Second)

**Review by psrikanth88 — CHANGES_REQUESTED** [review]

File: event-schema/event_schema.go (hunk 1 above)
Comment (achettyiitr) [reply event-schema/event_schema.go]: Done.

File: event-schema/event_schema.go (hunk 2 above)
Comment (achettyiitr) [reply event-schema/event_schema.go]: Done.

**Review by psrikanth88 — APPROVED** [review]

**Pull request merged**